autowarefoundation/autoware
# When using use_sukf in imm_ukf_pda_track, behavior_state is wrong

## Bug fix

### Fixed bug


Bug report is here. https://github.com/autowarefoundation/autoware_ai/issues/645
When using `use_sukf` in `imm_ukf_pda_track`, it expects output's `behavior_state` to be `1`, but currently it has `2`.

### Fix applied


#### Approach
Modified function `ImmUkfPda::updateBehaviorState` so that it could work with `use_sukf` param.

#### How to check the applied fix
1. Download Moriyama rosbag. [link](https://drive.google.com/file/d/19TKN05e7z0MUWFApXrFpiuQq2bAjTQ3s/view?usp=sharingl)

2. Open runtime manager

3. Play the rosbag in `simulation` tab.

4. Click `app` next to `imm_ukf_pda_track` in `computing` tab.

5. Select  `input topic` to  `detection/lidar_detector/objects`

6. Check `use_sukf`

7. Click OK and close `app` configuration.
![image](https://user-images.githubusercontent.com/10920881/57355818-de082c00-71a9-11e9-956d-b39ba8ac7513.png)

8. In a terminal, `$rostopic echo /detection/objects/objects[0]/behavior_state`

9. `1` is shown.
